### PR TITLE
Refactor `function-calc-no-unspaced-operator` for coverage improvements

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

None.

> Is there anything in the PR that needs further explanation?

In addition to coverage, this renames variables/functions for more readability.

### CODE REVIEW

File: lib/rules/function-calc-no-unspaced-operator/index.mjs [hunk 1]
```diff
@@ -219,46 +147,30 @@ const rule = (primary, _secondaryOptions, context) => {
 							operator,
 						);
 					}
-				} else if (charAfter && charAfter !== ' ') {
```
Comment (Mouvedia) [inline lib/rules/function-calc-no-unspaced-operator/index.mjs:222]: I am surprised that the removal of this block doesn't trigger a regression.
Is this now covered by something else or was it always unnecessary?

Comment (ybiquitous) [reply lib/rules/function-calc-no-unspaced-operator/index.mjs:222]: Yes, this is already covered by:

- https://github.com/stylelint/stylelint/blob/9d5b9f095309e4f957205f7c297f7c83aba02e8f/lib/rules/function-calc-no-unspaced-operator/index.mjs#L113
- https://github.com/stylelint/stylelint/blob/9d5b9f095309e4f957205f7c297f7c83aba02e8f/lib/rules/function-calc-no-unspaced-operator/index.mjs#L119
- https://github.com/stylelint/stylelint/blob/9d5b9f095309e4f957205f7c297f7c83aba02e8f/lib/rules/function-calc-no-unspaced-operator/index.mjs#L138

E.g.

```css
calc(-1px)    /* operatorIndex === 0 */
calc(1px-2)   /* charBefore === 'x' && charAfter === '2' */
calc(1px- 2)  /* charBefore === 'x' */
```

Comment (Mouvedia) [reply lib/rules/function-calc-no-unspaced-operator/index.mjs:222]: https://github.com/stylelint/stylelint/blob/45d9a2d5656caefc8f276082cc0b5a4a43ada230/lib/rules/function-calc-no-unspaced-operator/__tests__/index.mjs#L371

seems to be passing indeed.

Comment (Mouvedia) [reply lib/rules/function-calc-no-unspaced-operator/index.mjs:222]: No wait there's an error in the tests :

https://github.com/stylelint/stylelint/blob/45d9a2d5656caefc8f276082cc0b5a4a43ada230/lib/rules/function-calc-no-unspaced-operator/__tests__/index.mjs#L373

It should be `expectedOperatorAfterSign` right?

Comment (Mouvedia) [reply lib/rules/function-calc-no-unspaced-operator/index.mjs:222]: https://github.com/stylelint/stylelint/blob/45d9a2d5656caefc8f276082cc0b5a4a43ada230/lib/rules/function-calc-no-unspaced-operator/__tests__/index.mjs#L371-L373

seems to be passing indeed.

**Comment by Mouvedia** [comment]
off-topic

Nothing to do with your refactor but maybe you could fix it in another PR; it seems that:

```css
a { top: calc(1px +-2px); }
```

expects `messages.expectedOperatorBeforeSign('+')` and not `messages.expectedAfter('+')`.
`calc(1px + -2px)` is valid and could be justified as consistency with the surrounding.
i.e. all the other calc uses `+` so it's visually cleaner

**Comment by ybiquitous** [comment]
Ah, good catch. This is a [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKAuEAO8ViQBswAKARmwA8UBqAWgCYqBKAbhQF8QAaEAMwCWhGADkkAWziIYlCdmEA6MAGdlPcBAB2ggOYIQwADqaUKQyBkYYmqMvN4A2sdOnzyjAE9hhAZox1IbQEdOnckGyQAJyhzZxQAXWMuXkDdADEISPEkDH0AK2UtdVhsNUQjEzMQdy8YHz97KsIcmHdzbji3T29ff1Tg0IxwqCiYhCaWtpAkkA4gA) to reproduce the incorrect message.

image

I'll try fixing it in the following PR 👍🏼

**Comment by ybiquitous** [comment]
@Mouvedia If you have no more problems or questions, I'll merge this PR.

**Pull request merged**